File: sets/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main()
{
    double index, L, S;

    //Getting user input
    string text = get_string("Text: ");

    //counting letters, words and sentences
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    //Calculating index
    L = ((double) letters / words) * 100.0;
    S = ((double) sentences / words) * 100.0;
    index = 0.0588 * L - 0.296 * S - 15.8;



    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }

    return 0;

}

int count_sentences(string text)
{
    int length = strlen(text);
    int periods = 0;

    for (int i = 0; i < length; i++)
    {
        if(text[i] =='.' || text[i] == '!' || text[i] == '?')
        {
            periods++;
        }
    }
    //periods are the same as sentences
    return periods;
}


int count_words(string text)
{
    int length = strlen(text);
    int spaces = 0;
    int words;

   for (int i = 0; i < length; i++)
    {
        if(isspace(text[i]))
        {
                spaces++;
        }
    }
    words = spaces + 1;
    return words;
}



//Function to count letters
int count_letters(string text)
{
    int length = strlen(text);
    int special = 0;
    int letters;

   for (int i = 0; i < length; i++)
    {
            if(!isalpha(text[i]))
            {
                special++;
            }
    }
    letters = length - special;
    return letters;
}

File: sets/pset6/readability/readability.py
from cs50 import get_string


def main():

    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Using Coleman-Liau index to calculate reading level
    L = (float(letters) / words) * 100.0
    S = (float(sentences) / words) * 100.0
    index = 0.0588 * L - 0.296 * S - 15.8

    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {round(index)}")


# function to count sentences
def count_sentences(text):
    periods = 0
    for i in text:
        if i == "." or i == "?" or i == "!":
            periods += 1
    return periods


# function to count words
def count_words(text):
    spaces = 0
    for i in text:
        if i == " ":
            spaces += 1
    words = spaces + 1
    return words


# function to count letters
def count_letters(text):
    special = 0
    for i in text:
        if not i.isalpha():
            special += 1
    letters = len(text) - special
    return letters


if __name__ == "__main__":
    main()
